json_rpc/clients/httpclient.nim

```diff
@@ -45,16 +45,34 @@ method call*(client: RpcHttpClient, name: string,
     req = HttpClientRequestRef.post(client.httpSession,
                                     client.httpAddress.get,
                                     body = reqBody.toOpenArrayByte(0, reqBody.len - 1))
-    res = await req.send()
+    res =
+      try:
+        await req.send()
+      except CancelledError as exc:
+        raise exc
+      except HttpError as exc:
+        raise exc
+      except AsyncStreamError as exc:
+        raise exc
+
+  let resStatusStr = $res.status
```

Comment:
Using `$` on the numeric status value will require a dynamic memory allocation. The code would be more efficient if it performs the check directly over the numeric value by using operators such as `<` and `>`.

Reply:
f4217a82c96546cecd35a777ff7b4a9314e7a4c6